akka-docs/src/main/paradox/stream/operators/RestartSource/onFailuresWithBackoff.md

```diff
@@ -14,11 +14,50 @@ Wrap the given @apidoc[Source] with a @apidoc[Source] that will restart it when
 
 ## Description
 
- This @apidoc[Source] will never emit a failure, since the failure of the wrapped @apidoc[Source] is always handled by
- restarting. The wrapped @apidoc[Source] can be cancelled by cancelling this @apidoc[Source].
- When that happens, the wrapped @apidoc[Source], if currently running will be cancelled, and it will not be restarted.
- This can be triggered simply by the downstream cancelling, or externally by introducing a @apidoc[KillSwitch] right
- after this @apidoc[Source] in the graph.
+Wraps the given @apidoc[Source] with a @apidoc[Source] that will restart it when it fails using an exponential backoff.
+The backoff resets back to `minbackOff` if there hasn't been a failure within `minBackoff`.
+ 
+This @apidoc[Source] will never emit a failure, since the failure of the wrapped @apidoc[Source] is always handled by
+restarting. The wrapped @apidoc[Source] can be completed by completing this @apidoc[Source].
+When that happens, the wrapped @apidoc[Source], if currently running will be cancelled, and it will not be restarted.
+This can be triggered by the downstream cancelling, or externally by introducing a @ref[KillSwitch](../../stream-dynamic.md#controlling-stream-completion-with-killswitch) right
+after this @apidoc[Source] in the graph.
+
+See also: 
+ 
+* @ref:[RestartFlow.onFailuresWithBackoff](../RestartFlow/onFailuresWithBackoff.md)
+* @ref:[RestartFlow.withBackoff](../RestartFlow/withBackoff.md)
+* @ref:[RestartSink.withBackoff](../RestartSink/withBackoff.md)
+
+## Examples
+
+This shows that a Source is not restarted if it completes, only if it fails. Tick is only printed
+three times as the `take(3)` means the inner source.
+
+Scala
+:  @@snip [Restart.scala](/akka-docs/src/test/scala/docs/stream/operators/source/Restart.scala) { #restart-failure-inner-complete }
+
+Java
+:  @@snip [Restart.java](/akka-docs/src/test/java/jdocs/stream/operators/source/Restart.java) { #restart-failure-inner-complete }
+
+If the inner source instead fails, it will be restarted with a backoff. The source emits 1, 2, 3, and then throws an exception.
```

Comment:
```suggestion
If the inner source instead fails, it will be restarted with an increasing backoff. The source emits 1, 2, 3, and then throws an exception.
```